# Apple
**Publication Details:** Title: Integrating genomics and multiplatform metabolomics enables metabolite QTL detection in breeding-relevant apple germplasm

Authors: Emma A. Bilbrey1, Kathryn Williamson2, Emmanuel Hatzakis2, Diane Doud Miller3, Jonathan Fresnedo-Ramírez3, Jessica L. Cooperstone1,2,\*

1 Department of Horticulture and Crop Science, The Ohio State University, Columbus, OH, USA, 43210

2 Department of Food Science and Technology, The Ohio State University, Columbus, OH, USA, 43210

3 Department of Horticulture and Crop Science, The Ohio State University, Wooster, OH, USA, 44691

\*Corresponding Author: 2001 Fyffe Court, Columbus, OH 43210, USA; Tel: +614 2922843; Email: [cooperstone.1\@osu.edu](mailto:cooperstone.1@osu.edu)

Journal: New Phytologist 
Year: 2021

DOI: https://doi.org/10.1111/nph.17693
Also available as a pre-print on biorXiv: https://doi.org/10.1101/2021.02.18.431481

- Apple (Malus × domestica) has commercial and nutritional value, but breeding constraints of tree crops limit varietal improvement. Marker-assisted selection minimises these drawbacks, but breeders lack applications for targeting fruit phytochemicals. To understand genotype–phytochemical associations in apples, we have developed a high-throughput integration strategy for genomic and multiplatform metabolomics data.
- Here, 124 apple genotypes, including members of three pedigree-connected breeding families alongside diverse cultivars and wild selections, were genotyped and phenotyped. Metabolite genome-wide association studies (mGWAS) were conducted with c. 10 000 single nucleotide polymorphisms and phenotypic data acquired via LC–MS and 1H NMR untargeted metabolomics. Putative metabolite quantitative trait loci (mQTL) were then validated via pedigree-based analyses (PBA).
- Using our developed method, 519, 726 and 177 putative mQTL were detected in LC–MS positive and negative ionisation modes, and NMR, respectively. mQTL were indicated on each chromosome, with hotspots on linkage groups 16 and 17. A chlorogenic acid mQTL was discovered on chromosome 17 via mGWAS and validated with a two-step PBA, enabling discovery of novel candidate gene–metabolite relationships.
- Complementary data from three metabolomics approaches and dual genomics analyses increased confidence in validity of compound annotation and mQTL detection. Our platform demonstrates the utility of multiomic integration to advance data-driven, phytochemical-based plant breeding.

To understand genotype-phytochemical associations in apple fruit, we have developed a high-throughput integration strategy for genomic and multi-platform metabolomics data.

**Context**: 124 apple genotypes, including members of three pedigree-connected breeding families alongside diverse cultivars and wild selections, were genotyped and phenotyped. Metabolite genome-wide association studies (mGWAS) were conducted with \~10,000 single nucleotide polymorphisms alongside phenotypic data acquired via liquid chromatography mass spectrometry (LC-MS) and 1H nuclear magnetic resonance (NMR) untargeted metabolomics. Putative metabolite quantitative trait loci (mQTL) were then validated via pedigree-based analyses (PBA).

The data that support the findings of this study are openly available in the supplemental information or in online repositories. Raw full scan LC–MS data can be found in MetaboLights at www.ebi.ac.uk/metabolights/MTBLS2327 (Haug et al., 2020). Data-dependent LC–MS/MS raw spectra can be found in the MassIVE database at 10.25345/C5B21P. All data used in analysis can be found in the Supporting Information. Code for data processing, analysis and visualisation is available here.

Code is divided into 3 parts.  The first is dedicated to data visualization and prep for mGWAS analyses. Data inputs can be found in the supplemental material for the publication. Certain outputs from this section are then used in the batch scripts for conducting mGWAS analyses.  The second is batch scripts used for mGWAS analyses are also provided in this repository as part 2. The third is dedicated to mGWAS and PBA results processing and visualization. Data inputs can be found in the supplemental material for the publication.
